test/functional/helpers/reloadPage.js

```diff
@@ -10,8 +10,20 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import { ClientFunction } from "testcafe";
+import { t, ClientFunction } from "testcafe";
 
-export default ClientFunction(() => {
-  document.location.reload();
+const getCurrentUrl = ClientFunction(() => {
+  return document.location.href;
 });
+
+export default async () => {
+  const currentUrl = await getCurrentUrl();
+  // navigateTo waits for the server to respond after a redirect occurs,
+  // which is why we use it instead of just calling document.location.reload()
+  // in our client function.
+  // We have to navigate to a different page and then back to the current page,
+  // because if we just tried to navigate to the same page we're on, TestCafe
+  // would hang.
+  await t.navigateTo("blank.html");
```

Comment:
I considered just loading `about:blank`, which works for this case, but then I thought about changing everywhere we were using `blank.html` to use `about:blank`, and there was one scenario that didn't work, so I decided I would just roll with `blank.html` (blank.html is a file we serve alongside our test page).